### PR TITLE
fix(multi-select): add fieldset, legend, and make boxes tabbable

### LINKED ISSUE #1346: ♿Multi-select AVT1 - August 2018♿
## Environment
Chrome Version 70.0.3538.67 (Official Build) (64-bit)
macOS Mojave 10/Windows 10

### General & Inline
Groups of radio buttons must be contained within a fieldset element.
screen shot 2018-10-25 at 5 35 10 pm

[The id referenced in the for attribute of each checkbox is not a valid input. This applies to every instance of the checkbox.
](https://aat.w3ibm.mybluemix.net/token/b62ec93e-0a1f-4e82-b96a-18e75818d205/51d217d2-9b4e-4e3e-a04e-66809340d9d0/latest///doc/w3/help/en-US/idhi_accessibility_check_g1062.html)
> This could be a DAP false positive, but it looks like we're nesting the input inside the label tag. Which might be why `for=` isn't properly hitting it's target. [Here's some more info that might be relevant.](https://stackoverflow.com/a/774073)
screen shot 2018-10-25 at 5 35 10 pm

## How to reproduce
1. [Download the Dynamic Accessibility Plugin tool here](https://ibm.ent.box.com/notes/335005856253?s=mhpgjlgss354mcidsv6rvpuu0bxlmfm4)
2. Run the extension using the August 2018 ruleset

### PR DESCRIPTION
**Pull request by dakahn**
Closes #1346

#### Changelog
- Changes container element around checkbox list from `div` to `fieldset`.
- Adds a `legend` child element to `fieldset`
- Removes duplicate `id` on container element for the checkbox list
- Removes `tabindex="-1"` so the checkboxes can be tabbed to
  _This seemed deliberate and there might be something I'm missing that requires the negative `tabindex`_

### CODE REVIEW

**Review by jnm2377 — APPROVED** [review]
Keyboard navigation works great with these changes!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @dakahn!

File: src/components/multi-select/multi-select.hbs [hunk 1]
```diff
@@ -21,17 +21,18 @@
       </svg>
     </div>
   </div>
-  <div class="{{@root.prefix}}--list-box__menu">
+  <fieldset class="{{@root.prefix}}--list-box__menu">
+    <legend class="{{@root.prefix}}--visually-hidden">Description of form elements within the fieldset</legend>
```
Comment (joshblack) [inline src/components/multi-select/multi-select.hbs]: Could we use `bx--assistive-text` here? The classes are equivalent but I believe the intended usage may be different.

File: src/components/multi-select/multi-select.hbs [hunk 2]
```diff
@@ -21,17 +21,18 @@
...
     {{#each items}}
-    <div class="{{@root.prefix}}--list-box__menu-item" id="{{id}}">
+    <div class="{{@root.prefix}}--list-box__menu-item">
       <div class="{{@root.prefix}}--form-item {{@root.prefix}}--checkbox-wrapper">
-        <label for="{{id}}" title="{{label}}" class="{{@root.prefix}}--checkbox-label">
-          <input type="checkbox" name="Option 1" readonly="" tabindex="-1" class="{{@root.prefix}}--checkbox" id="{{id}}" value="on">
+        <label title="{{label}}" class="{{@root.prefix}}--checkbox-label">
```
Comment (joshblack) [inline src/components/multi-select/multi-select.hbs:29]: Given that we remove `for` here, how are we linking the `<label>` to the `<input>`?

Comment (dakahn) [reply src/components/multi-select/multi-select.hbs:29]: The label is implied by wrapping the input. Which should be totally cool with screen reader tech. I'll make a note to specifically make sure this works as intended in AVT2 testing.

**Pull request merged**